File: unity/Runtime/Core/Internal/IPlatformImpl.cs
using System.Threading.Tasks;

namespace EE.Internal {
    public interface IPlatformImpl {
        bool IsApplicationInstalled(string applicationId);
        bool OpenApplication(string applicationId);
        string GetApplicationId();
        string GetApplicationName();
        string GetVersionName();
        string GetVersionCode();
        string GetSha1Signature();
        bool IsInstantApp();
        bool IsTablet();
        float GetDensity();
        (int, int) GetViewSize();
        (int, int) GetScreenSize();
        Task<string> GetDeviceId();
        bool SendMail(string recipient, string subject, string body);
        Task<bool> TestConnection(string hostName, float timeOut);
        Task<string> GetInstallReferrerUrl();
    }
}

File: src/ts/src/core/internal/IPlatformImpl.ts
export interface IPlatformImpl {
    isApplicationInstalled(applicationId: string): boolean;
    openApplication(applicationId: string): boolean;
    getApplicationId(): string;
    getApplicationName(): string;
    getVersionName(): string;
    getVersionCode(): string;
    getSha1Signature(): string;
    isInstantApp(): boolean;
    isTablet(): boolean;
    getDensity(): number;
    getViewSize(): [number, number];
    getScreenSize(): [number, number];
    getDeviceId(): Promise<string>;
    sendMail(recipient: string, subject: string, body: string): boolean;
    testConnection(hostName: string, timeOut: number): Promise<boolean>;
}